File: Proyecto/src/main/java/API/Editar.java
package API;

import java.io.IOException;
import java.io.PrintWriter;
import javax.servlet.ServletException;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.Statement;

public class Editar extends HttpServlet {

    private PrintWriter out;

    @Override
    protected void doGet(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {
        out = response.getWriter();
        String id = request.getParameter("id");
        String pre = request.getParameter("pregunta");
        String res = request.getParameter("respuesta");
        String ima11 = request.getParameter("ima11");
        String val11 = request.getParameter("val11");
        String ima12 = request.getParameter("ima12");
        String val12 = request.getParameter("val12");
        String ima13 = request.getParameter("ima13");
        String val13 = request.getParameter("val13");
        String ima14 = request.getParameter("ima14");
        String val14 = request.getParameter("val14");
        String ima21 = request.getParameter("ima21");
        String val21 = request.getParameter("val21");
        String ima22 = request.getParameter("ima22");
        String val22 = request.getParameter("val22");
        String ima23 = request.getParameter("ima23");
        String val23 = request.getParameter("val23");
        String ima24 = request.getParameter("ima24");
        String val24 = request.getParameter("val24");
        try {
            Class.forName("com.mysql.cj.jdbc.Driver");
            Connection enlace = DriverManager.getConnection("jdbc:mysql://localhost/crudjson", "root", "1234");
            Statement consulta = enlace.createStatement();

            int rs = consulta.executeUpdate("update tablajson set columnajson = json_replace(columnajson,'$.pregunta','"
                    + pre + "','$.respuesta','" + res + "','$.drags[0].valor','" + val11 + "','$.drags[0].imagen','"
                    + ima11 + "'," +
                    "'$.drags[1].valor','" + val12 + "','$.drags[1].imagen','" + ima12 + "','$.drags[2].valor','"
                    + val13 + "','$.drags[2].imagen','" + ima13 + "','$.drags[3].valor','" + val14
                    + "','$.drags[3].imagen','" + ima14 + "'," +
                    "'$.targets[0].valor','" + val21 + "','$.targets[0].imagen','" + ima21 + "','$.targets[1].valor','"
                    + val22 + "','$.targets[1].imagen','" + ima22 + "','$.targets[2].valor','" + val23
                    + "','$.targets[2].imagen','" + ima23 + "'," +
                    "'$.targets[3].valor','" + val24 + "','$.targets[3].imagen','" + ima24 + "') where idEjercicio = "
                    + id + ";");

            if (rs != 0) {
                System.out.println("yes");
                out.println("yes");
            } else {
                out.println("no");
            }
        } catch (Exception e) {
            e.printStackTrace();
        }

    }
}

File: reactBD/src/components/editar.js
import React from "react";
import { Redirect } from "react-router-dom"
import { Button, Container } from "react-bootstrap";
import axios from "axios";
import { Link } from "react-router-dom";
import Swal from 'sweetalert2';
class Info extends React.Component {

    state = {
        id: "",
        pregunta: "",
        respuesta: "",
        drags: [],
        targets: []
    }

    componentDidMount() {
        const qId = new URLSearchParams(window.location.search).get("id");
        if (qId) {
            axios.get("Pregunta?id=" + qId).then(response => {
                const question = response.data[0];
                this.setState({ ...question });
            }).catch(error => {
                console.info(error);
                alert("Ha ocurrido un error");
            });
        }
    }
    validar = (pregunta, respuesta, imagen11, valor11, imagen12, valor12, imagen13, valor13, imagen14, valor14, imagen21, valor21, imagen22, valor22, imagen23, valor23, imagen24, valor24) => {
        //fetch('http://localhost:8080/Proyecto/Login?User='+usuario+'&password='+password+'')
        const preid = new URLSearchParams(window.location.search).get("id");
        fetch('Editar?id=' + preid + '&pregunta=' + pregunta + '&respuesta=' + respuesta + '&ima11=' + imagen11 + '&val11=' + valor11 + '&ima12=' + imagen12 + '&val12=' + valor12 + '&ima13=' + imagen13 + '&val13=' + valor13 +
            '&ima14=' + imagen14 + '&val14=' + valor14 + '&ima21=' + imagen21 + '&val21=' + valor21 + '&ima22=' + imagen22 + '&val22=' + valor22 + '&ima23=' + imagen23 + '&val23=' + valor23 + '&ima24=' + imagen24 + '&val24=' + valor24)
            .then(response => response.text())
            .then(usuario => {
                let ret = usuario.includes("yes");
                console.log(ret);

                if (ret) {
                    Swal.fire(
                        'Actualizado',
                        'Updated',
                        'success'
                    )
                }
                else {
                    Swal.fire(
                        'No se pudo actualizar',
                        'Error',
                        'error'
                    )

                }
            })

    }


    render() {
        const { pregunta, respuesta, drags, targets } = this.state;
        return (
            <Container className="MarginContainer">
                <div className="card-header text-center">
                    <h3>Editar Pegunta</h3>
                </div>
                <div className="card-body">
                    <form action="">
                        <label className="form-label">Pregunta :  </label>
                        <input className="form-control" type="text" defaultValue={pregunta} id="pregunta" />
                        <label className="form-label">Respuesta : </label>
                        <input className="form-control" type="text" defaultValue={respuesta} id="respuesta" />
                        <div className="row">
                            <label className="form-label"><b>Drag options</b> </label>
                        </div>


                        <div className="row">
                            {

                                drags.map((option, index) => {

                                    return (
                                        <div className="col">
                                            <label className="form-label">link {index + 10} : <input className="form control" type="text" defaultValue={option.imagen} id={index + 10}></input></label>
                                            <img src={option.imagen} className="ImageContainer" />
                                            <label className="form-label">Valor {index + 20} : <input className="form-control" type="text" defaultValue={option.valor} id={index + 20}></input></label>

                                        </div>
                                    );
                                })

                            }
                        </div>
                        <label className="form-label"><b>Target Options</b></label>

                        <div className="row">
                            {
                                targets.map((target, index) => {
                                    let valor = 0;
                                    return (
                                        <div className="col">
                                            <label className="form-label">link {index} : {valor} <input className="form-control" type="text" defaultValue={target.imagen} id={index + 30}></input></label>

                                            <img src={target.imagen} className="ImageContainer" />
                                            <label className="form-label">valor : <input className="form-control" type="text" defaultValue={target.valor} id={index + 40}></input></label>

                                        </div>

                                    )
                                }
                                )
                            }
                        </div>
                        <div className="d-grid m-3">
                            <Button variant="success" onClick={() => this.validar(
                                document.getElementById("pregunta").value,
                                document.getElementById("respuesta").value,
                                document.getElementById("10").value,
                                document.getElementById("20").value,
                                document.getElementById("11").value,
                                document.getElementById("21").value,
                                document.getElementById("12").value,
                                document.getElementById("22").value,
                                document.getElementById("13").value,
                                document.getElementById("23").value,
                                document.getElementById("30").value,
                                document.getElementById("40").value,
                                document.getElementById("31").value,
                                document.getElementById("41").value,
                                document.getElementById("32").value,
                                document.getElementById("42").value,
                                document.getElementById("33").value,
                                document.getElementById("43").value
                            )} >
                                ACTUALIZAR
                            </Button>
                        </div>

                    </form>

                    <div className="d-grid m-3">
                        <Button
                            variant="success"
                            className="M-6">
                            <Link to={`/Proyecto/home`} className="link-light" >
                                HOME
                            </Link>
                        </Button>
                    </div>

                </div>
            </Container>
        )
    }
}

export default Info;